### DISCUSSION TITLE
*Is stack allocated memory zero initiallized >

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/fsharp#14972] [Q&A] [answered]
NativeInterop.NativePtr.stackalloc

**Comment by vzarytovskii** [comment, marked as answer]
It just emits `localloc` instruction, and here's what spec says:

> The localloc instruction allocates size (type native unsigned int) bytes from the local dynamic memory pool and returns the address (a managed pointer, type &) of the first allocated byte. The block of memory returned is initialized to 0 only if the initialize flag on the method is true (see Partition I). The area of memory is newly allocated. When the current method returns the local memory pool is available for reuse.

So, in other words it depends whether method has `localsinit`.


F# doesn't skip locals init by default iirc.

**Reply by Xyncgas** [reply]
Thanks
